ntoskrnl/ke/i386/mp.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture specific source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
+} APInfo;
+
+/* FUNCTIONS *****************************************************************/
+VOID
+NTAPI
+KeStartAllProcessors()
+{
+     /* Prep some globals for the AP */
+    PVOID KernelStack;
+    PVOID DPCStack;
+    KPROCESSOR_STATE ProcessorState;
+    KDESCRIPTOR GdtDesc, IdtDesc;
+
+    HighestPhysicalAddress.QuadPart = UINT64_MAX;
+    __sgdt(&GdtDesc.Limit);
+    __sidt(&IdtDesc.Limit);
+
+    /* Attempt to allocate memory for the new setup */
+    SIZE_T APInfoSize = sizeof(struct APInfo) + 2;
+    PVOID PAPInfo = MmAllocateContiguousMemory(APInfoSize, HighestPhysicalAddress);
```

Comment:
If you think these fields won't be used in AP boot code, that should be a simple NonPagedPool

Reply:
And it should be allocated for each CPU.